### DISCUSSION TITLE
State of HTML 2023

### DISCUSSION THREAD

**Discussion by SachaG** [Devographics/surveys#111] [State of HTML Design]
Discussing a potential State of HTML survey.

**Comment by chrishtr** [comment]
I think a State of HTML survey for 2023 would be great. It would be a place to ask questions like developer needs and opinions regarding:

* Missing HTML features
* Missing DOM APIs
* Events & observation patterns
* Accessibility & content semantics
* Editing
* Selection
* Behavior of built-in form and navigation control elements
* Template rendering and state management
* JS extensions to the rendering pipeline

This survey would also finish surveying the core technologies of the web, which are HTML, CSS and JavaScript.

**Reply by LeaVerou** [reply]
Hi @chrishtr, btw here are the features we're considering for State of HTML: https://coda.io/@leaverou/state-of-html
These are the features for Part 1 of the survey, which measures awareness (and this year, hopefully also [sentiment](https://github.com/Devographics/surveys/discussions/183)). If there are any you had in mind when compiling the list above that are missing, please let me know ASAP.

**Reply by chrishtr** [reply]
I'd like to also ask about:

1. Seamless iframes
2. An API to move elements within the DOM without losing focus, animation or reloading iframes or video

Thanks!

**Reply by LeaVerou** [reply]
Hi @chrishtr, 

Thanks for sharing.



> 1. Seamless iframes

Since there was syntax for this (`<iframe seamless>`) we *could* ask about this in as a Feature question (the questions that measure awareness/usage/sentiment specific features), though asking about a removed feature gives me a bit of pause. I filed https://github.com/Devographics/surveys/discussions/202 to discuss
We also have a pain points [question](https://coda.io/d/_dTGBFYq175J/External-content_su8K_#_lueSL) around this focus area, which might suffice?

> 2. An API to move elements within the DOM without losing focus, animation or reloading iframes or video

Since this is not an existing specced feature, we an't ask about it as a feature question. However, we also already have a [question](https://coda.io/d/_dTGBFYq175J/Interactivity_suv46#_luvQE) about pain points around its focus area (DOM mainpulation), which should hopefully bring up these pain points without priming respondents by mentioning them directly. Maybe that would suffice? 

Looking forward to your thoughts.

**Comment by chrishtr** [comment]
The current proposal for seamless iframes looks like [this](https://github.com/w3c/csswg-drafts/issues/1771). I'd most like to know whether developers encounter this need or not, not the specifics of the API.

For the second one, I agree that a question around the pain points is also the best feedback point.

**Reply by LeaVerou** [reply]
I see, in that case I think the targeted pain points questions are more likely to demonstrate the need for seamless iframes as well (e.g. I suspect many people will ask for some way to include templates in HTML, which is one of the main use cases for seamless iframes).

Here are some existing questions that I think may uncover such a need:
- [What are your biggest pain points around using external content in HTML?](https://coda.io/d/_dTGBFYq175J/External-content_su8K_#_lueSL)
- [Which existing HTML features are you unable to use?](https://coda.io/d/_dTGBFYq175J/Missing-features-and-pain-points_suOQH#_lutn1)
- [Which components do you find yourself using across multiple projects?](https://coda.io/d/_dTGBFYq175J/Missing-features-and-pain-points_suOQH#_lupoC)
- [If you could add 3 elements to HTML, which would you add?](https://coda.io/d/_dTGBFYq175J/Missing-features-and-pain-points_suOQH#_luzbz)

Do you think these suffice for this? If not, what would you suggest?

**Reply by chrishtr** [reply]
How about:

Do you see any gaps in HTML support for responsive design? If so, which?
